app/views/questionnaire.py

```diff
@@ -280,15 +301,24 @@ def post_block_handler(
             return redirect(next_url)
 
     questionnaire_store_updater.save_answers(form)
-    next_location = path_finder.get_next_location(current_location=current_location)
 
-    if _is_end_of_questionnaire(block, next_location):
+    next_location = path_finder.get_next_location(
+        current_location, get_current_section(user=current_user)
+    )
```

Comment:
Why does `get_next_location` need to know the current section?

Reply:
It needs the section to pass to `routing_path` - it doesn't need it as an arg though, as the section can be determined from the `current_location`

Reply:
Getting it from `current_location` would be better